src/benchmarklib/benchmark_runner.cpp

```diff
@@ -93,20 +77,18 @@ BenchmarkRunner::~BenchmarkRunner() {
 void BenchmarkRunner::run() {
   std::cout << "- Starting Benchmark..." << std::endl;
 
-  const auto available_queries_count = _query_generator->available_query_count();
-  _query_plans.resize(available_queries_count);
-  _query_results.resize(available_queries_count);
+  const auto available_item_count = _benchmark_item_runner->available_item_count();
+  _results.resize(available_item_count);
```

Comment:
Using a vector-as-a-map here doesn't work, right? Because `available_item_count` would be 3 for `selected_items()` `{1, 15, 16}` but would be accessed with an index of `16` later, right? I made `_results` an undered_map

Reply:
ah, we allocated 22 slots and then only filled in those that are in selected_items(), I see.. Added a bit of commenting to the _results declaration in the header